### DISCUSSION TITLE
Getting response for the first time if the component is rendered multiple times

### DISCUSSION THREAD

**Discussion by gopusrikanth04** [TanStack/query#2054] [Q&A]
Hello All,

```
import ValueHelpInput from './../../valuehelp/ValueHelpInput';


         <ValueHelpInput
              serviceName="MasterService"
              entityName="Customers"
              returnField="title"
              onChange={() => {}}
            />
```

![issue2](https://user-images.githubusercontent.com/47174113/112991541-ddedec80-9184-11eb-8c14-9c5a60e9f999.png)

In my functional component I am using using react-query to get the data

using that component in table row. but the react query is called and get response for the first time only
and the next time the react query is not called(key is same in each time)

![issue](https://user-images.githubusercontent.com/47174113/112991419-b6971f80-9184-11eb-9af8-7ff5e9d55d68.png)

How to solve his problem 
thank you

**Comment by TkDodo** [comment]
sorry, I don't quite follow. It's very hard to tell anything without a code sample that includes the react-query call - a minimal reproducible codesandbox example would be best.

I also don't get what you mean with:

>  the react query is not called(key is same in each time)

react-query does not fetch data on every re-render. also not sure if you are talking about query-keys of react-query or `key` of a react component 🤔 🤷

**Reply by gopusrikanth04** [reply]
Hello Dominik,
The issue is I have a reusable component input ( it has a suggestion list which is read from backed ). Now this input is used as a table column. lets say i have added 5 records to the table,  for the 1st record the suggestions are read from backend which is absolutely correct. But for  the second record onwards data is not been read ( may be due to its not getting invalidated and getting from the cache  ) which is its functionality. now the second record input is not having the suggestion data. 

We have done this before using state and it used to work completely fine. when we implemented this with React-query it stopped working and not sure weather its the issue with react-query.

**Reply by TkDodo** [reply]
a codesandbox example would be really helpful to understand the issue :)

**Reply by gopusrikanth04** [reply]
I am very sorry. The issue was from our side. In Query function the data was set to state and that state data was used for suggestions for input. As react-query was not getting invalidated from second time onwards the state was not set.

thanks and regards.
Srikanth  Gopu.
